packages/web3-eth-accounts/src/index.js

```diff
@@ -162,6 +172,9 @@ Accounts.prototype.signTransaction = function signTransaction(tx, privateKey, ca
             if (transaction.accessList) {
                 // EIP-2930
                 transaction.type = "0x01"
+            } else if (transaction.maxPriorityFeePerGas || transaction.maxFeePerGas) {
```

Comment:
I think this should be the first case, since in practice 1559 can be appended to any transaction type.

Reply:
But what happens if they specify `accessList` and `maxFeePerGas`?